apis/kueue/v1beta1/resourceflavor_types.go

```diff
@@ -65,6 +65,7 @@ type ResourceFlavorSpec struct {
 	// +optional
 	// +listType=atomic
 	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:validation:XValidation:rule="self.all(x, x.effect in ['NoSchedule', 'PreferNoSchedule', 'NoExecute'])", message="supported taint effect values: 'NoSchedule', 'PreferNoSchedule', 'NoExecute'"
 	NodeTaints []corev1.Taint `json:"nodeTaints,omitempty"`
```

Comment:
@jpbetz do you think there is anything better we can do here to validate a Taint?

I don't think we could add kubebuilder tags to `corev1.Taint`, or can we?

Or should we create our own type to mimic `Taint` and add kubebuilder rules there?

Reply:
(Using this CEL check until kubebuilder respects `// enum` seems like a good stop gap though..)

Reply:
Thanks!

For now, opened an issue for kubebuilder https://github.com/kubernetes-sigs/kubebuilder/issues/3861